python/lsst/daf/persistence/posixStorage.py

```diff
@@ -601,6 +606,22 @@ def search(root, path, searchParents=False):
             else:
                 return None
 
+    @staticmethod
+    def storageExists(uri):
+        """Ask if a storage at the location described by uri exists
+
+        Parameters
+        ----------
+        root : string
+            URI to the the root location of the storage
+
+        Returns
+        -------
+        bool
+            True if the storage exists, false if not
+        """
+        return os.path.exists(PosixStorage._pathFromURI(uri))
```

Comment:
It looks like this boils down to a `stat(2)` call, so this should be nice and atomic.